src/MultivariateStats.jl

```diff
@@ -441,12 +443,224 @@ function MLJBase.predict(m::BayesianLDA, (core_res, class_list, priors), Xnew)
     return [MLJBase.UnivariateFinite(class_list, P[j, :]) for j in 1:size(P, 1)]
 end
 
+####
+#### BayesianSubspaceLDA
+####
+
+"""
+BayesianSubspaceLDA(; kwargs...)
+
+$BayesianSubspaceLDA_DESCR
+
+## Parameters
+
+* `normalize=true`: Option to normalize Sb by number of observations in each class, one of `true` or `false`.
+* `out_dim`: the output dimension, i.e dimension of the transformed space, automatically set if 0 is given (default).
+* `priors=nothing`: if `priors = nothing` estimates the prior probabilities from the data, else it uses the user specified `UnivariateFinite` prior probabilities.
+See also the [package documentation](https://multivariatestatsjl.readthedocs.io/en/latest/lda.html).
+For more information about the algorithm, see the paper by Li, Zhu and Ogihara, [Using Discriminant Analysis for Multi-class Classification: An Experimental Investigation](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.89.7068&rep=rep1&type=pdf).
+"""
+@mlj_model mutable struct BayesianSubspaceLDA <: MLJBase.Probabilistic
+    normalize::Bool=true
+    out_dim::Int   = 0::(_ ≥ 0)
+    priors::Union{Nothing, MLJBase.UnivariateFinite} = nothing
+end
+
+function MLJBase.fit(model::BayesianSubspaceLDA, ::Int, X, y)
+    class_list = MLJBase.classes(y[1]) #class list containing unique entries in y
+    nclasses   = length(class_list)
+
+    # NOTE: copy/transpose
+    Xm_t   = MLJBase.matrix(X, transpose=true) # now p x n matrix
+    yplain = MLJBase.int(y) # vector of n ints in {1,..., nclasses}
+    p, n   = size(Xm_t)
+
+    # check output dimension default is min(p, nc-1)
+    def_outdim = min(p, nclasses - 1)
+    # if unset (0) use the default; otherwise try to use the provided one
+    out_dim = ifelse(model.out_dim == 0, def_outdim, model.out_dim)
+    # check if the given one is sensible
+    out_dim ≤ def_outdim || throw(ArgumentError("`out_dim` must not be larger than `min(p, nc-1)` where `p` is the dimension of `X` and `nc` is the number of classes."))
+
+    # check to make sure sample size is greater or equals nclasses
+    n ≥ nclasses || throw(ArgumentError("The number of samples is less than the number of classes"))
+    
+    ## Estimates prior probabilities is unspecified by user.
+    if model.priors == nothing
+        priors = proportions(yplain)
+    else
+        #check if the length of priors is same as nclasses
+        size(MLJBase.classes(model.priors)) == nclasses || throw(ArgumentError("Invalid size of `priors`"))
+        priors = MLJBase.pdf.(model.priors, class_list)
+    end
+
+    # # Code gotten from JuliaStats/MultivariateStats.jl
+    # # Compute centroids, class weights, and deviation from centroids
+    # # Note Sb = Hb*Hb', Sw = Hw*Hw'
+    cmeans, cweights, Hw = MS.center(Xm_t, Int.(yplain), nclasses)
+    dmeans = cmeans .- (model.normalize ? MS.mean(cmeans, dims=2) : cmeans * (cweights / (n)))
```

Comment:
just checking: did you adjust the dimensions here? was it dims=1 in the MultivariateStats code? // did you test that this version of the code leads to identical results? (just checking due to possible issues with the fact that MVS uses p x n convention instead of our n x p

Reply:
I didn't adjust the dims. I has already transposed the matrix to be p x n

Reply:
The results gave identical results

Reply:
thanks I should have seen  this :)